File: solutions/2244.py
from functools import reduce
from itertools import groupby
from typing import List


class Solution:

    def minimumRounds(self, tasks: List[int]) -> int:
        groups = [len(list(group)) for key, group in groupby(sorted(tasks))]
        if groups.count(1):
            return -1
        return reduce(lambda x, y: x + y // 3 + ((y % 3) != 0), groups, 0)

File: solutions/2244.cpp
#include <unordered_map>
#include <vector>

class Solution {
 public:
  int minimumRounds(std::vector<int> &tasks) {
    std::unordered_map<int, std::size_t> counts;
    for (auto const &task : tasks) ++counts[task];

    int rounds = 0;
    for (auto const &[_, count] : counts)
      if (count > 1)
        rounds += count / 3 + ((count % 3) != 0);
      else
        return -1;

    return rounds;
  }
};
